### PR TITLE
`ptr()` clean-up

### PR DESCRIPTION
**Pull request by mkatliar**
- Make `ptr()` implementation independent of specific matrix and vector types
- Blaze-specific code moved to `blast/blaze`

### CODE REVIEW

**Review by roversch — COMMENTED** [review]
I can see that you changed the interface a bit (`ptr(*A)` vs `ptr(A)`), which I don't understand the reasons for. Could you elaborate?

File: include/blast/math/dense/Getf2.hpp [hunk 1]
```diff
@@ -106,7 +103,7 @@ namespace blast
     template <typename MT>
     inline void getf2(DenseMatrix<MT, columnMajor>& A, size_t * ipiv)
     {
-        getf2(rows(*A), columns(*A), ptr(A), ipiv);
+        getf2(rows(*A), columns(*A), ptr(*A), ipiv);
```
[the remark this answers is not in the document]
Comment (mkatliar) [reply include/blast/math/dense/Getf2.hpp:106]: No, `*A` is not dereferencing a pointer. `DenseMatrix<MT, ...>` is a CRTP base for `MT`, and `DenseMatrix<MT, ...>::operator*()` casts `DenseMatrix<MT, ...>` to `MT`.

File: include/blast/math/dense/Ger.hpp [hunk 2]
```diff
@@ -185,9 +169,9 @@ namespace blast
         using ET = ElementType_t<MT1>;
         size_t constexpr TILE_SIZE = TileSize_v<ET>;
 
-        BLAZE_CONSTRAINT_MUST_BE_SAME_TYPE(ElementType_t<VT0>, ET);
-        BLAZE_CONSTRAINT_MUST_BE_SAME_TYPE(ElementType_t<VT1>, ET);
-        BLAZE_CONSTRAINT_MUST_BE_SAME_TYPE(ElementType_t<MT0>, ET);
+        // BLAZE_CONSTRAINT_MUST_BE_SAME_TYPE(ElementType_t<VT0>, ET);
+        // BLAZE_CONSTRAINT_MUST_BE_SAME_TYPE(ElementType_t<VT1>, ET);
+        // BLAZE_CONSTRAINT_MUST_BE_SAME_TYPE(ElementType_t<MT0>, ET);
```
[the remark this answers is not in the document]
Comment (mkatliar) [reply include/blast/math/dense/Ger.hpp:174]: Removed!

File: include/blast/math/dense/Getf2.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (roversch) [reply include/blast/math/dense/Getf2.hpp:106]: :+1:

**Review by roversch — APPROVED** [review]

**Comment by mkatliar** [comment]
> I can see that you changed the interface a bit (`ptr(*A)` vs `ptr(A)`), which I don't understand the reasons for. Could you elaborate?

This is because `ptr()` expects the most derived type like `blaze::StaticMatrix<...>` and not the CRTP base `blaze::DenseMatrix<...>`. But, I just realized that I could implement
```c++
template <typename MT, bool SO>
inline ptr(blaze::DenseMatrix<MT, SO>& m)
{
    return ptr(*m);
 }
 ```
 and then the `*`s would not be needed.

**Pull request merged**